Cargo.toml

```diff
@@ -115,6 +115,7 @@ tempfile = "3.9"
 serde_json = "1.0" # Used in documentation tests.
 bstr = "1.9.0"
 tempfile = "3.9"
+itertools = "0.13.0"
```

Comment:
Secondary: unless it's impossible to do without, please don't add a new dependency. If you don't know how to do the same thing without `itertools`, it's fine. I'll send a follow-up to remove the dep.

Reply:
Out of curiosity: why the hard line about dependencies (even dev dependencies, like this one)?

Reply:
Didn't realize it was a dev-dependency. It's ok in this case. For dependencies it's because MSRV is not considered a breaking-change, which in turn broke `sysinfo` multiple times. It's very frustrating but not something that I can fix unless I enforce the version of all dependencies I use in `sysinfo`. So instead, I keep the number of dependencies I use as low as possible.

Also, it's better for clean builds to have a small number of deps (although not very important as clean builds are not that common when using a project).